### DISCUSSION TITLE
[Enhancement] Allow embedding .NET MAUI screens/apps into native apps

### DISCUSSION THREAD

**Discussion by akamud** [dotnet/maui#322] [Ideas]
## Summary
Make it possible to have applications in Swift/Objective-C and Java/Kotlin consume apps, libs and screens built with .NET MAUI.

## Intended Use Case
Creating an app or screen in .NET MAUI and being able to embed it into native apps (Swift/Objective-C or Java/Kotlin) is a big feature from frameworks like [React Native](https://reactnative.dev/docs/integration-with-existing-apps). 
I work at a consulting company and many customers ask for a way to do this. A common scenario is that they already have a legacy app that they don't plan to throw away all at once, or they want to transition to a new app/architecture in cycles, allowing them to make incremental releases.
Another common scenario is when the customer want to try a new technology. So they can do this with React Native today, they can take their existing apps and build a feature with it to validate the development cycle using a new technology. 

We've seen many opportunities of using Xamarin today being discarded for those reasons, so having a way of doing this would make it more easier for people to try .NET MAUI.

What I'm asking is more on the lines of [Embeddinator-4000](https://github.com/mono/Embeddinator-4000), but I don't see it as a project that can be taken to production today (maybe I'm wrong?)

I understand that maybe this should be in the Xamarin original repo, and not on .NET MAUI itself? Please tell me if I should move this. I think it is worth at least having the discussion here.

**Comment by SkyeHoefling** [comment]
I prototyped using the Embeddinator-4000 to build a Xamarin.Forms plugin that could be consumed by native iOS and Android projects. It does work, but I don't recommend it. The Embeddinator-4000 is a really powerful tool and it is still in preview. I would be worried about not being able to generate new builds as iOS and Android make updates.

Adding support for something like this will open up doors for teams that build plugins for other app developers. When I had to do this we were building a SDK that iOS and Android app developers could use to make their integrations with our enterprise system easier. The gave us a consistent look and feel when 3rd parties integrated but allowed the 3rd parties to use their own custom apps.

**Reply by ILAgent** [reply]
Any update? IMHO it's necessary possibility for modern crossplatform framework.

**Reply by binsys** [reply]
Any update?

**Reply by ILAgent** [reply]
friendly ping

**Reply by iSeiryu** [reply]
Embeddinator-4000 repo was archived :(
React-native has an ability to run RN apps/screens within an existing Android/iOS app via https://expo.dev/
It would be awesome to have a similar ability with MAUI!

**Comment by Majestic-Mustang** [comment]
This is neat and can help accelerate MAUI adoption. Please consider doing this. Thanks!
@davidortinau
